### ISSUE TITLE
TypeError: sbot._flumeUse is not a function

### ISSUE THREAD

**Issue by enko** [ssbc/ssb-names#1]
I tried to use this module but it failed me:

```
const ssbClient = util.promisify(window.require('ssb-client'));
this.bot = await ssbClient();
const Names = window.require('ssb-names');
const names = Names.init(this.bot);
```

This resulted in this stack trace:

```
TypeError: sbot._flumeUse is not a function
    at Object.exports.init (C:\Users\tim\Datenknoten\dev\ngx-ssb-client\node_modules\ssb-names\index.js:75)
    at ScuttlebotService.<anonymous> (scuttlebot.service.ts:45)
```

This seem related: https://github.com/ssbc/ssb-links/issues/2

Could you please have a look at it?

**Comment by enko** [comment]
I have to add that I use `ssb-party` to get a client instance, maybe that is the problem here, but I do not know.

**Comment by enko** [comment]
I have tried latest `ssb-client` and it is still the same. I'm running patchwork as my main server, maybe that is a problem?

**Comment by dominictarr** [comment]
oh, you need to add it on the server side not the client side.
patchfoo also depends on plugins, and explains the installation proceduce here:
https://github.com/ssbc/patchfoo#install-extras

**Comment by enko** [comment]
> oh, you need to add it on the server side not the client side.

Then I run into the next problem:

```
λ sbot plugins.install ssb-backlinks
Command not found: plugins.install
```

I think this comes, that the sbot-server by patchwork has not the plugins module installed? I run patchwork to have a sbot server, so I do not have to start a server.

**Comment by dominictarr** [comment]
oh yeah, it loads these plugins:

https://github.com/ssbc/patchwork/blob/master/server-process.js#L7-L23

sbot by default comes with some slightly different plugins, also loads a plugins-plugin like this: https://github.com/ssbc/scuttlebot/blob/master/bin.js#L41-L56

If you are developing your own client, I'd recommend just running sbot on it's own, and then bundling it into a single program like patchwork, patchbay, and ticktack do once are happy with it.

**Comment by enko** [comment]
Here I come again 😄 

I finaly integrated scuttlebot directly into the client and am able to use ssb-names, but `getSignifier` and `getImageFor` does not return an object like indicated [in the documentation](https://github.com/ssbc/ssb-names/blob/master/README.md), but just a name and a blob id.

Am I doing something wrong here?

**Comment by dominictarr** [comment]
no, you arn't doing anything wrong. the documentation is completely out of date

**Comment by dominictarr** [comment]
fixed in 3.1.2

**Comment by enko** [comment]
Thanks a lot!
